mono/tests/Makefile.am

```diff
@@ -707,116 +861,9 @@ DISABLED_TESTS=			\
-DISABLED_TESTS_CI_PR =	\
-	appdomain-threadpool-unload.exe	\
-	appdomain-thread-abort.exe
-
-DISABLED_TESTS_CI=	\
-	$(DISABLED_TESTS)	\
-	$(shell if [ "x$(CI_PR)" != "x" ]; then echo $(DISABLED_TESTS_CI_PR); fi)	\
-	main-returns-background-resetabort.exe \
-	main-returns-background-abort-resetabort.exe	\
-	thread6.exe	\
-	assemblyresolve_event3.exe \
-	delegate2.exe	\
-	finally_guard.exe \
-	gc-altstack.exe \
-	generic-xdomain.2.exe
-
-AOT_DISABLED_TESTS=constraints-load.exe
-
-# These only compile with MS CSC
-TEST_CSC_SRC=			\
-	vararg.cs
-
-# constraints-load.il: 
-# Failed to load method 0x6000007 from '..../mono/tests/constraints-load.exe' due to 
-# Could not resolve type with token 01000002 assembly:mscorlib, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089 type:System.BrokenIComparable`1 member:<none>.
-IL_SRC_MOBILE_STATIC_NOT_SUPPORTED=	\
-	constraints-load.il \
-	bug-515884.il
-
-TEST_IL_SRC_UNIVERSAL=			\
-	field-access.il		\
-	method-access.il	\
-	ldftn-access.il		\
-	cpblkTest.il		\
-	vbinterface.il		\
-	calliTest.il		\
-	calliGenericTest.il		\
-	ckfiniteTest.il		\
-	fault-handler.il		\
-	locallocTest.il		\
-	initblkTest.il		\
-	qt-instance.il		\
-	vararg.il		\
-	bug-29859.il		\
-	bug-78549.il		\
-	static-fields-nonconst.il	\
-	reload-at-bb-end.il	\
-	test-enum-indstoreil.il	\
-	filter-bug.il		\
-	even-odd.il		\
-	bug-82022.il	\
-	vt-sync-method.il	\
-	enum_types.il	\
-	invalid-token.il	\
-	call_missing_method.il	\
-	call_missing_class.il	\
-	ldfld_missing_field.il	\
-	ldfld_missing_class.il	\
-	find-method.2.il	\
-	bug-79215.2.il	\
-	bug-79956.2.il	\
-	bug-327438.2.il \
-	bug-387274.2.il	\
-	bug-426309.2.il	\
-	ldtoken_with_byref_typespec.2.il \
-	resolve_method_bug.2.il \
-	resolve_field_bug.2.il \
-	resolve_type_bug.2.il	\
-	generics-sharing-other-exc.2.il	\
-	generic-ldobj.2.il	\
-	generic-mkrefany.2.il	\
-	generic-refanyval.2.il	\
-	generic-ldtoken.2.il	\
-	generic-ldtoken-method.2.il	\
-	generic-ldtoken-field.2.il	\
-	generic-tailcall.2.il	\
-	generic-tailcall2.2.il	\
-	generic-array-exc.2.il	\
-	generic-valuetype-newobj2.2.il	\
-	generic-valuetype-newobj.2.il	\
-	generic-constrained.2.il	\
-	generic-type-load-exception.2.il	\
-	bug-81466.il	\
-	bug457574.il	\
-	bug445361.il	\
-	bug-463303.il	\
-	bug469742.2.il	\
-	bug-528055.il	\
-	array_load_exception.il \
-	bug-481403.il	\
-	interface-with-static-method.il	\
-	bug-633291.il	\
-	delegate-with-null-target.il	\
-	bug-318677.il	\
-	gsharing-valuetype-layout.il	\
-	invalid_generic_instantiation.il \
-	bug-45841-fpstack-exceptions.il
-
-if INSTALL_MOBILE_STATIC
-TEST_IL_SRC= \
-	$(TEST_IL_SRC_UNIVERSAL)
-else
-TEST_IL_SRC= \
-	$(TEST_IL_SRC_MOBILE_STATIC_NOT_SUPPORTED) \
-	$(TEST_IL_SRC_UNIVERSAL)
-endif
+	$(PROFILE_DISABLED_TESTS) \
+	$(shell if [ "x$(AOT)" != "x" ]; then echo $(AOT_DISABLED_TESTS); fi) \
+	$(shell if [ "x$(CI)" != "x" ]; then echo $(CI_DISABLED_TESTS); if [ "x$(CI_PR)" != "x" ]; then echo $(CIPR_DISABLED_TESTS); fi; fi)
```

Comment:
I'd split this line into two, I could image wanting to exclude the CI_PR tests but not the CI tests or vice versa when running locally. Plus it looks nicer :)